src/file/tokio_compat_file.rs

```diff
@@ -469,6 +474,24 @@ impl AsyncWrite for TokioCompatFile {
             .map(|n| min(n, max_write_len))
             .unwrap_or(max_write_len);
 
+        let write_limit = self.get_auxiliary().tokio_compat_file_write_limit();
+        let write_len = self.project().write_len;
+        match write_len.checked_add(n as usize) {
+            Some(new_write_len) if new_write_len > write_limit => {
+                ready!(self.as_mut().poll_flush(cx))?;
+            }
+            None => {
+                // overflow. This has to be a separate cases since
```

Comment:
Improvement to my own comment:

```suggestion
                // case overflow
                // This has to be a separate cases since
```